keps/sig-scheduling/20190317-non-preempting-priorityclass.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Add NonPreempting Option For PriorityClasses
+authors:
+  - "@vllry"
+owning-sig: sig-scheduling
+participating-sigs:
+  - sig-scheduling
+reviewers:
+  - "k82cn"
+  - "wgliang"
+approvers:
+  - "bsalamat"
+editor: Vallery Lancey
+creation-date: 2019-03-17
+last-updated: 2019-03-17
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Allow PriorityClasses To Be Non-Preempting
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+
+
+## Summary
+
+[PriorityClasses](https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/) are a GA feature as on 1.14,
+which impact the scheduling and eviction of pods.
+Pods will be scheduled according to descending priority.
+If a pod cannot be scheduled due to insufficient resources,
+lower-priority pods will be descheduled ("preempted") to make room.
+
+This proposal makes the preempting behavior optional for a PriorityClass,
+by adding a new field to PriorityClasses.
+If a PriorityClass does not have preemption enabled,
+a pod of that PriorityClass will not trigger preemption of other pods.
+
+## Motivation
+
+High-priority, non-preempting workloads are a common data science use case.
+Preempting batch workloads is a waste, as the work unit must be repeated.
```

Comment:
I think this could be explained a bit more clearly. This sentence is fine, but after it I would add something like "In general, there may be workloads that are not time-sensitive to schedule (it's OK for them to wait if the cluster is full), but that once running, it is important not to preempt them."